core/src/main/java/org/testcontainers/utility/RegistryAuthLocator.java

```diff
@@ -206,7 +209,29 @@ private AuthConfig runCredentialProvider(String hostName, String credHelper) thr
                 .execute()
                 .outputUTF8()
                 .trim();
-        } catch (Exception e) {
+        }
+        catch (InvalidResultException e) {
+            if(e.getResult() != null && e.getResult().hasOutput()) {
+                // ErrCredentialsNotFound standardizes the not found error, so every helper returns the same message
+                // We can handle it properly with fallback to other resources.
+                // https://github.com/docker/docker-credential-helpers/blob/19b711cc92fbaa47533646fa8adb457d199c99e1/credentials/error.go#L4-L6
+                if ("credentials not found in native keychain".equals(e.getResult().outputString().trim())) {
```

Comment:
I would suggest checking only "credentials not found", "native keychain" says something about the implementation and might change in future

Also, what about this one?
https://github.com/docker/docker-credential-helpers/blob/d499cf5cb90029fdd50c550243b16363857694d7/osxkeychain/osxkeychain_darwin.go#L21

To be honest, I feel uneasy about these string checks. That's just a text without an error code or something, and (knowing Docker guys) it might be changed anytime.

Is it possible to perform a forward check instead of relying on the message only?

Reply:
I agree, that just text isn't good.
I could change to check the content of "credentials not found".

In osxkeychain, they translate it to general message as used here:
https://github.com/docker/docker-credential-helpers/blob/d499cf5cb90029fdd50c550243b16363857694d7/osxkeychain/osxkeychain_darwin.go#L90-L91

Central/common credentials error messages
https://github.com/docker/docker-credential-helpers/blob/19b711cc92fbaa47533646fa8adb457d199c99e1/credentials/error.go#L24-L28

Reply:
:-) will see how it comes out